src/orca-loader/Hyper.res

```diff
@@ -331,10 +331,14 @@ let make = (publishableKey, options: option<JSON.t>, analyticsInfo: option<JSON.
                 }
                 postSubmitMessage(dict)
 
-                if (
-                  isSdkButton ||
-                  (val->JSON.Decode.bool->Option.getOr(false) && redirect === "always")
-                ) {
+                if isSdkButton {
+                  if !(val->JSON.Decode.bool->Option.getOr(false)) {
+                    resolve1(json)
+                  } else {
+                    resolve1(data)
+                    Window.replace(returnUrl)
+                  }
+                } else if val->JSON.Decode.bool->Option.getOr(false) && redirect === "always" {
                   Window.replace(returnUrl)
                 } else if !(val->JSON.Decode.bool->Option.getOr(false)) {
```

Comment:
```
val->JSON.Decode.bool->Option.getOr(false)
```
this is repeated couple of times here. Can you define it in a variable on top (with a clear name) and use it in the 3 places?

Reply:
got it, did the change